File: apps/mobile/ios/Sources/FlagshipUI/Screens/AddServerChooserScreen.swift
import SwiftUI

/// Two-card chooser used everywhere we need the "add a server" decision:
///   - Onboarding (when the user has zero servers)
///   - In-app: Home pane → Add server, Settings → Add server
///
/// Both branches lead to existing flows (`CreateServerStubScreen` /
/// `PodPairScreen`). The screen is presentation-only — the parent owns
/// the navigation push.
public struct AddServerChooserScreen: View {
    @Environment(\.colorScheme) private var scheme
    public enum Mode { case onboarding, inApp }

    let mode: Mode
    var onProvision: () -> Void
    var onPair: () -> Void

    public init(
        mode: Mode = .inApp,
        onProvision: @escaping () -> Void = {},
        onPair: @escaping () -> Void = {}
    ) {
        self.mode = mode
        self.onProvision = onProvision
        self.onPair = onPair
    }

    public var body: some View {
        let c = FSColors.scheme(scheme)
        ScrollView {
            VStack(alignment: .leading, spacing: FS.space.s6) {
                Spacer().frame(height: mode == .onboarding ? FS.space.s12 : FS.space.s4)
                VStack(alignment: .leading, spacing: FS.space.s2) {
                    Text(title).font(FS.font.h2()).foregroundColor(c.text)
                    Text(subtitle).font(FS.font.body()).foregroundColor(c.textMuted)
                }

                Button(action: onProvision) {
                    chooserCard(
                        icon: "server.rack",
                        accent: c.primary,
                        title: "Provision a new box",
                        body: "Mint a build code, download a personalized Alpine ISO, flash it to commodity hardware. Cert + tunnel come up automatically.",
                        c: c
                    )
                }
                .buttonStyle(.plain)

                Button(action: onPair) {
                    chooserCard(
                        icon: "qrcode.viewfinder",
                        accent: c.success,
                        title: "Pair an existing box",
                        body: "Already have a Flagship server running somewhere? Scan its pairing QR or paste the 6-character code.",
                        c: c
                    )
                }
                .buttonStyle(.plain)

                Spacer().frame(height: FS.space.s12)
            }
            .padding(.horizontal, FS.space.s6)
        }
        .background(c.bg.ignoresSafeArea())
        .navigationTitle(mode == .onboarding ? "" : "Add a server")
        .navigationBarTitleDisplayMode(.inline)
    }

    private var title: String {
        mode == .onboarding ? "Get your first server." : "Add a server."
    }
    private var subtitle: String {
        mode == .onboarding
            ? "Pick a path — both end with your stuff running on hardware you control."
            : "Add another box to your fleet. Each one is independent."
    }

    private func chooserCard(icon: String, accent: Color, title: String, body: String, c: FSColors) -> some View {
        FSCard(padding: FS.space.s6) {
            VStack(alignment: .leading, spacing: FS.space.s3) {
                ZStack {
                    RoundedRectangle(cornerRadius: FS.radius.sm)
                        .fill(accent.opacity(0.12))
                    Image(systemName: icon)
                        .foregroundColor(accent)
                        .font(.system(size: 22, weight: .semibold))
                }
                .frame(width: 44, height: 44)
                Text(title).font(FS.font.h3()).foregroundColor(c.text)
                Text(body).font(FS.font.body()).foregroundColor(c.textMuted)
                HStack {
                    Spacer()
                    Image(systemName: "arrow.right")
                        .foregroundColor(accent)
                }
            }
        }
    }
}

File: apps/mobile/android/app/src/main/java/com/flagshipserver/app/ui/screens/AddServerChooserScreen.kt
// Two-card chooser used everywhere we ask "add a server": onboarding
// (zero servers yet) + the in-app + button on Home / Settings. Both
// branches navigate to existing flows. The screen is presentation-only;
// the caller owns the nav.

package com.flagshipserver.app.ui.screens

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.PaddingValues
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.outlined.QrCode2
import androidx.compose.material.icons.outlined.Storage
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.platform.testTag
import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.flagshipserver.app.ui.components.FSCard
import com.flagshipserver.app.ui.components.FSGhostButton
import com.flagshipserver.app.ui.theme.FS

enum class AddServerMode { ONBOARDING, IN_APP }

@Composable
fun AddServerChooserScreen(
    mode: AddServerMode = AddServerMode.IN_APP,
    onProvision: () -> Unit,
    onPair: () -> Unit,
    onCancel: (() -> Unit)? = null,
) {
    val scroll = rememberScrollState()
    Column(
        Modifier
            .fillMaxSize()
            .verticalScroll(scroll)
            .padding(horizontal = FS.space.s6),
    ) {
        Spacer(Modifier.height(if (mode == AddServerMode.ONBOARDING) FS.space.s12 else FS.space.s4))
        Text(
            if (mode == AddServerMode.ONBOARDING) "Get your first server." else "Add a server.",
            color = FS.colors.text,
            style = TextStyle(fontSize = 28.sp, lineHeight = 36.sp, fontWeight = FontWeight.Medium),
        )
        Text(
            if (mode == AddServerMode.ONBOARDING) {
                "Pick a path — both end with your stuff running on hardware you control."
            } else {
                "Add another box to your fleet. Each one is independent."
            },
            color = FS.colors.textMuted,
            style = TextStyle(fontSize = 14.sp, lineHeight = 20.sp),
        )
        Spacer(Modifier.height(FS.space.s4))
        ChooserCard(
            icon = Icons.Outlined.Storage,
            accent = FS.colors.primary,
            title = "Provision a new box",
            body = "Mint a build code, then use the Flagship burner to write it to a USB stick and boot commodity hardware. Cert + tunnel come up automatically.",
            cta = "Provision →",
            onClick = onProvision,
            testTag = "chooser-provision",
        )
        Spacer(Modifier.height(FS.space.s3))
        ChooserCard(
            icon = Icons.Outlined.QrCode2,
            accent = FS.colors.success,
            title = "Pair an existing box",
            body = "Already have a Flagship server running somewhere? Scan its pairing QR or paste the 6-character code.",
            cta = "Pair →",
            onClick = onPair,
            testTag = "chooser-pair",
        )
        if (onCancel != null) {
            Spacer(Modifier.height(FS.space.s4))
            FSGhostButton(label = "Cancel", onClick = onCancel, block = true)
        }
        Spacer(Modifier.height(FS.space.s12))
    }
}

@Composable
private fun ChooserCard(
    icon: ImageVector,
    accent: androidx.compose.ui.graphics.Color,
    title: String,
    body: String,
    cta: String,
    onClick: () -> Unit,
    testTag: String? = null,
) {
    FSCard(
        padding = PaddingValues(FS.space.s6),
        modifier = if (testTag != null) Modifier.testTag(testTag) else Modifier,
    ) {
        Column(verticalArrangement = Arrangement.spacedBy(FS.space.s3)) {
            Box(
                modifier = Modifier
                    .size(44.dp)
                    .clip(RoundedCornerShape(FS.radius.sm))
                    .padding(0.dp),
                contentAlignment = Alignment.Center,
            ) {
                Box(
                    Modifier
                        .size(44.dp)
                        .clip(RoundedCornerShape(FS.radius.sm))
                        .padding(0.dp),
                )
                Icon(icon, contentDescription = title, tint = accent)
            }
            Text(title, color = FS.colors.text, style = TextStyle(fontSize = 18.sp, fontWeight = FontWeight.SemiBold))
            Text(body, color = FS.colors.textMuted, style = TextStyle(fontSize = 14.sp, lineHeight = 20.sp))
            FSGhostButton(label = cta, onClick = onClick, block = true)
        }
    }
}
